client/pushToServer.go

```diff
@@ -22,79 +21,31 @@ func pushFolderToServer(conn net.Conn, folderPath string) {
 
 		if entry.IsDir() {
 			// If the entry is a directory, recursively call pushFolderToServer
-			pushFolderToServer(conn, entryPath)
+			err := sendFile(entryPath, conn)
```

Comment:
De nesting hier kan minder. Of is dit een functie welke hierna nog doorloopt? Ik kan het namelijk niet goed zien in de diff van Github😅

`if entry.IsDir() {
    err := sendFile(entryPath, conn)
	if err != nil {
		return
                // of een print van de error aangezien het nu dubbel is.
	}
return
}
// If the entry is a file, push it to the server
err := sendFile(entryPath, conn)
if err != nil {
	return
}

// Add a delimiter between files (except for the last file)
if i < len(entries)-1 {
	_, err := conn.Write([]byte("\n\n"))
	if err != nil {
		return
		}
}`

Reply:
Volgens mij was de nesting minder maar zijn er meer toegevoegd door de Go error handling.

Reply:
Eerder was het een if else en daarin nog een if. thats all maar door de error handling is het een stuk meer geworden..